ext/sqlite3/database.c

```diff
@@ -717,6 +717,55 @@ static VALUE transaction_active_p(VALUE self)
   return sqlite3_get_autocommit(ctx->db) ? Qfalse : Qtrue;
 }
 
+static int callback_function(void *callback_ary, int count, char **data, char **columns)
+{
+  VALUE * result = (VALUE *)callback_ary;
+  VALUE row_ary = rb_ary_new();
+  int i;
+
+  for (i = 0; i < count; i++) {
+    if (data[i] == NULL){
+      rb_ary_push(row_ary, Qnil);
+    } else {
+      VALUE str = rb_str_new_cstr(data[i]);
```

Comment:
This means that return values will always be strings where the normal query mechanism will typecast.  For example "SELECT 1" will return the string "1" with this, but the integer 1 using other APIs.  I'm not sure that this kind of inconsistency is a good thing.